src/shogun/machine/IterativeLinearMachine.cpp

```diff
@@ -0,0 +1,69 @@
+/*
+* This software is distributed under BSD 3-clause license (see LICENSE file).
+*
+* Authors: Shubham Shukla
+*/
+
+#include <shogun/base/progress.h>
+#include <shogun/lib/common.h>
+#include <shogun/lib/config.h>
+#include <shogun/machine/IterativeLinearMachine.h>
+
+using namespace shogun;
+
+CIterativeLinearMachine::CIterativeLinearMachine() : CLinearMachine()
+{
+	m_current_iteration = 0;
+	m_complete = false;
+
+	SG_ADD(
+	    &m_current_iteration, "current_iteration",
+	    "Current Iteration of training", MS_NOT_AVAILABLE);
+	SG_ADD(
+	    &m_max_iterations, "max_iterations", "Maximum number of Iterations",
+	    MS_AVAILABLE);
+	SG_ADD(&m_complete, "complete", "Convergence status", MS_NOT_AVAILABLE);
+}
+
+CIterativeLinearMachine::~CIterativeLinearMachine()
+{
+	SG_UNREF(m_continue_features);
+}
+
+bool CIterativeLinearMachine::train_machine(CFeatures* data)
+{
+	m_current_iteration = 0;
+	m_complete = false;
+	init_model(data);
+	return continue_train();
+}
+
+bool CIterativeLinearMachine::continue_train()
+{
+	reset_computation_variables();
+	auto pb = SG_PROGRESS(range(m_max_iterations));
+	while (m_current_iteration < m_max_iterations && !m_complete)
+	{
+		COMPUTATION_CONTROLLERS
+		iteration();
+		m_current_iteration++;
+		pb.print_progress();
+	}
+	pb.complete();
+
+	if (m_complete)
+		SG_INFO(
+		    "%s algorithm converged after %d iterations.\n", this->get_name(),
+		    m_current_iteration)
+	else
+		SG_WARNING(
+		    "%s algorithm did not converge after %d iterations.\n",
+		    this->get_name(), m_current_iteration)
```

Comment:
i used m_current_iteration here so that even if it is prematurely stopped the warning shows correct iterations.

Reply:
Question: I guess we dont want the warning to be shown if the user stopped prematurely. Only if the algorithm actually did that number of iterations, see what I mean?

Reply:
we dont want warning in case of premature stop ?

Reply:
This means, only showiung the warning if complete is false and m_current_iteration == max_iterations

Reply:
I think we should have one. But that should be different. It should just be an SG_INFO message that logs that the user chose to stop X prematurely after Y iterations

Reply:
but the CSignal will show something if we press CTRL+C

Reply:
Yeah or that. Just something that informs the user that he aborts after X iterations ...